README.md

```diff
@@ -108,25 +107,138 @@ file { '/etc/someservice/db.yaml':
+  ```sh-session
+  $ openssl s_client -showcerts -servername [DAP_INSTANCE_DNS_NAME] \
+    -connect [DAP_INSTANCE_DNS_NAME]:443 < /dev/null 2> /dev/null \
+    | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p'
+  -----BEGIN CERTIFICATE-----
+  ...
+  -----END CERTIFICATE-----
+  ```
+- `version`: Conjur API version, defaults to 5.
+
+_Note that not all variables are required for each method of configuration._
+
+#### Conjur host identity with API key
+
+The simplest way to get started with a Conjur application identity is to
+[create a host in Conjur](https://docs.conjur.org/Latest/en/Content/Operations/Policy/statement-ref-host.htm)
+and then provide its Conjur credentials to this module. There are a few ways to provide
+the Conjur Puppet module with these credentials and they are outlined in
+the following sections.
+
+##### Updating the Puppet manifest
+
+When you update the Puppet manifest to include the Conjur host identity and API key, you
+are configuring the Puppet **server** with the Conjur identity information.
+
+In this example, after you have created a Conjur host named `redis001`, you can add
+its host identity and its API key to your manifest like this:
 ```puppet
 class { 'conjur':
   appliance_url   => 'https://conjur.mycompany.com/',
+  account         => 'myorg',
   authn_login     => 'host/redis001',
   authn_api_key   => Sensitive('f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3'),
-  ssl_certificate => file('/abslute/path/to/conjur-ca.pem')
+  ssl_certificate => file('/absolute/path/to/conjur-ca.pem')
 }
 ```
 
-#### Special instructions for Windows hosts
+##### Using Hiera
+
+You can also add the Conjur identity configuration to Hiera, which provides the Conjur
+identity information to the Puppet **server**:
+
+```yaml
+---
+lookup_options:
+  '^conjur::authn_api_key':
+    convert_to: 'Sensitive'
+
+conjur::appliance_url: 'https://conjur.mycompany.com/'
+conjur::account: 'myorg'
+conjur::authn_login: 'host/redis001'
+conjur::authn_api_key: 'f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3'
+# conjur::cert_file: '/absolute/path/to/conjur-ca.pem'
+conjur::ssl_certificate: |
+  -----BEGIN CERTIFICATE-----
+  ...
+  -----END CERTIFICATE-----
+```
+
+##### Using Conjur identity files (Linux agents only)
+
+To configure **Linux agents** with a Conjur host identity, you can add the Conjur host
+and API key to
+[Conjur identity files](https://docs.conjur.org/Latest/en/Content/Get%20Started/key_concepts/machine_identity.html)
+`/etc/conjur.conf` and `/etc/conjur.identity`.
+
+Using the same `redis001` host as above, you would create a `conjur.conf` file that
+contains:
+```yaml
+---
+account: myorg
+plugins: []
+appliance_url: https://conjur.mycompany.com
+cert_file: "/absolute/path/to/conjur-ca.pem"
+```
+
+and a `conjur.identity` file that contains:
+```netrc
+machine conjur.mycompany.com
+    login host/redis001
+    password f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3
+```
+
```

Comment:
maybe we should add a note here that says that the module will automatically check for these files and use them if they are available, and (presumably?) no other configuration is provided?

Reply:
```suggestion

The Conjur Puppet Module will automatically check for these files on your node and use them if they
are available.
```